framework/src/systems/NonlinearSystemBase.C

```diff
@@ -2923,6 +2923,12 @@ NonlinearSystemBase::setPreconditioner(std::shared_ptr<MoosePreconditioner> pc)
   _preconditioner = pc;
 }
 
+const std::shared_ptr<const MoosePreconditioner>
+NonlinearSystemBase::getPreconditioner() const
```

Comment:
Ah! So this is what you were talking about yesterday. There's no reason to return a shared pointer here. You aren't sharing ownership. This would be a little cleaner if you just returned a const reference or a const pointer to the object inside of the shared_pointer instead of casting and returning a new shared pointer that you are only going to query.

Reply:
It can be null, there isn't always a MoosePreconditioner object to reference

Reply:
OK - then not a reference but a raw const pointer (not sharing ownership).

Reply:
I like that better.

Reply:
👍